xarray/core/combine.py

```diff
@@ -208,6 +219,8 @@ def _dataset_concat(datasets, dim, data_vars, coords, compat, positions):
     dim, coord = _calc_concat_dim_coord(dim)
     datasets = [as_dataset(ds) for ds in datasets]
     datasets = align(*datasets, join='outer', copy=False, exclude=[dim])
+    # TODO: compute dask coords with a single invocation of dask.compute()
```

Comment:
Yes, this would be even better :)

Reply:
I didn't bother as it would be a considerable complication and it would not do any good in the most common case where the "dask" variable is just a lazy load from disk